src/libopenrave-core/environment-core.h

```diff
@@ -4179,91 +4157,51 @@ class Environment : public EnvironmentBase
-                        if( ::tolower(filename[len-i-1]) != (*it)[it->size()-i-1] ) {
-                            bsuccess = false;
-                            break;
-                        }
-                    }
-                    if( bsuccess ) {
-                        return true;
-                    }
-                }
+        for (const std::string& ext : s_geometryextentsions) {
+            if (StringEndsWith(filename, ext)) {
+                return true;
             }
         }
         return false;
     }
 
-    static bool _IsJSONURI(const std::string& uri)
-    {
-        string scheme, authority, path, query, fragment;
-        string s1, s3, s6, s8;
-        static pcrecpp::RE re("^(([^:/?#]+):)?(//([^/?#]*))?([^?#]*)(\\?([^#]*))?(#(.*))?");
-        bool bmatch = re.FullMatch(uri, &s1, &scheme, &s3, &authority, &path, &s6, &query, &s8, &fragment);
-        return bmatch && scheme.size() > 0 && _IsJSONFile(path); //scheme.size() > 0;
-    }
-
     static bool _IsJSONFile(const std::string& filename)
     {
-        size_t len = filename.size();
-        if( len < 5 ) {
-            return false;
-        }
-        if( filename[len-5] == '.' && ::tolower(filename[len-4]) == 'j' && ::tolower(filename[len-3]) == 's' && ::tolower(filename[len-2]) == 'o' && ::tolower(filename[len-1]) == 'n' ) {
-            return true;
-        }
-        return false;
+        return StringEndsWith(filename, ".json");
     }
 
     static bool _IsJSONData(const std::string& data)
     {
         return data.size() >= 2 && data[0] == '{';
     }
 
-    static bool _IsMsgPackURI(const std::string& uri)
-    {
-        string scheme, authority, path, query, fragment;
-        string s1, s3, s6, s8;
-        static pcrecpp::RE re("^(([^:/?#]+):)?(//([^/?#]*))?([^?#]*)(\\?([^#]*))?(#(.*))?");
-        bool bmatch = re.FullMatch(uri, &s1, &scheme, &s3, &authority, &path, &s6, &query, &s8, &fragment);
-        return bmatch && scheme.size() > 0 && _IsMsgPackFile(path); //scheme.size() > 0;
-    }
-
     static bool _IsMsgPackFile(const std::string& filename)
     {
         // .msgpack
-        size_t len = filename.size();
-        if( len < 8 ) {
-            return false;
-        }
-        if( filename[len-8] == '.' && ::tolower(filename[len-7]) == 'm' && ::tolower(filename[len-6]) == 's' && ::tolower(filename[len-5]) == 'g' && ::tolower(filename[len-4]) == 'p' && ::tolower(filename[len-3]) == 'a' && ::tolower(filename[len-2]) == 'c' && ::tolower(filename[len-1]) == 'k' ) {
-            return true;
-        }
-        return false;
+        return StringEndsWith(filename, ".msgpack");
     }
 
     static bool _IsMsgPackData(const std::string& data)
     {
         return data.size() > 0 && !std::isprint(data[0]);
     }
 
+    static bool _IsEncryptedFile(const std::string& filename)
+    {
+        return StringEndsWith(filename, ".pgp") || StringEndsWith(filename, ".gpg");
```

Comment:
Looks like the code above says we only support `.gpg`, so let's remove `.pgp` here